server/events/vcs/gitlab_client.go

```diff
@@ -358,14 +358,12 @@ func (g *GitlabClient) UpdateStatus(repo models.Repo, pull models.PullRequest, s
 	if err != nil {
 		return err
 	}
-	// refTarget is set to current branch if no pipeline is assigned to the commit,
-	// otherwise it is set to the pipeline created by the merge_request_event rule
+	// refTarget is set to the current branch if no pipeline is assigned to the commit,
+	// otherwise it is set to the ref of the Head Pipeline
 	refTarget := pull.HeadBranch
-	if mr.Pipeline != nil {
-		switch mr.Pipeline.Source {
-		case "merge_request_event":
-			refTarget = fmt.Sprintf("refs/merge-requests/%d/head", pull.Num)
-		}
+	if mr.HeadPipeline != nil {
```

Comment:
Does this change also deprecate older gitlab versions

Reply:
`head_pipeline` was introduced in GitLab v12 back in 2020.

Reply:
Gitlab only support the current and previous two major versions, which is currently 16, 15 and 14, reference: [GitLab Satement of Support](https://about.gitlab.com/support/statement-of-support/#version-support)

Reply:
Thank you this is very helpful.